### PR TITLE
ref(trace) use event emitter to schedule events

### PR DESCRIPTION
**Pull request by JonasBa**
Split the rendering logic to react to events emitted by the scheduler (very similar to how profiling works). This will allow us to add independent components that can react to view changes and rerender their state however they need to and avoid react lifecycle updates where they are not needed.

By doing this we also fix a race condition where we failed to scroll to the row in the list (because we can await virtualized list initialization on initial render)

This prepares the UI work for minimap as the timeline indicators can subscribe to updates and are decoupled from the main span view

### CODE REVIEW

File: static/app/views/performance/newTraceDetails/trace.tsx [hunk 1]
```diff
@@ -206,6 +212,43 @@ export function Trace({
...
+    const onTraceSpaceChange: TraceEvents['initialize trace space'] = () => {
+      manager.recomputeTimelineIntervals();
+      manager.recomputeSpanToPXMatrix();
+      manager.draw();
+    };
+    const onDividerResize: TraceEvents['divider resize'] = view => {
+      manager.recomputeTimelineIntervals();
+      manager.recomputeSpanToPXMatrix();
+      manager.draw(view);
+    };
```
Comment (JonasBa) [inline static/app/views/performance/newTraceDetails/trace.tsx:235]: This is what we want the span view to react to. When it happens, we recompute the areas affected by timeline change and redraw the space

File: static/app/views/performance/newTraceDetails/traceRenderers/traceView.tsx [hunk 2]
```diff
@@ -1,8 +1,78 @@
 import {mat3} from 'gl-matrix';
 
+import clamp from 'sentry/utils/number/clamp';
+
 // Computes the transformation matrix that is used to render scaled
 // elements to the DOM and draw the view.
 export class TraceView {
```
Comment (JonasBa) [inline static/app/views/performance/newTraceDetails/traceRenderers/traceView.tsx:7]: This was moved from the manager since the view is a separate concept that is not related to the virtualized list.

File: static/app/views/performance/newTraceDetails/traceRenderers/virtualizedViewManager.tsx [hunk 3]
```diff
@@ -11,7 +11,9 @@ import type {
 } from 'sentry/views/performance/newTraceDetails/traceModels/traceTree';
```
Comment (JonasBa) [inline static/app/views/performance/newTraceDetails/traceRenderers/virtualizedViewManager.tsx:11]: A lot of the changes from this file is just code that was moved to TraceView and updates to access the view on the stored field

**Review by Abdkhan14 — APPROVED** [review]
